src/driver/compile.ml

```diff
@@ -225,7 +225,7 @@ let sherlodoc_index_one ~output_dir (index : Odoc_unit.index) =
   Sherlodoc.index ~format:`js ~inputs ~dst ();
   rel_path
 
-let html_generate output_dir linked =
+let html_generate ~occurrence_file output_dir linked =
```

Comment:
The index is set to be used for other purpose in the future (sidebar) so it doesn't make sense to build it from inside this function.
I think the `compile_index` code should be moved out of this function, `html_generate` should take `~db_path` instead.

Reply:
That this function calls `compile_index` was already true before this PR and isn't changed. Agreed that it'd be nicer to factor it out but let's not hold up this PR for that.